lm_eval/tasks/realtoxicityprompts/metric.py

```diff
@@ -26,6 +29,14 @@ def toxicity_perspective_api(references, predictions, **kwargs):
     url = f"https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key={API_KEY}"
 
```

Comment:
Setting QPS as environment variable, using QPS=1 as default.

```suggestion
    # Default Queries Per Second (QPS) from Perspective API is 1,
    # but you can request a higher QPS for your API key
    try:
        qps = int(os.environ["PERSPECTIVE_API_QPS"])
    except:
        eval_logger.warning(
            "Setting the Queries Per Second for PERSPECTIVE API to 1. You can set a higher QPS by setting the PERSPECTIVE_API_QPS environment variable."
        )
        qps = 1
    
    sleeping_time = 1 / qps
```

Reply:
In hindsight, let's not log a warning, as this is called for each item in the dataset.

Reply:
Then it feels that should be moved somewhere else, so we don't define this once every query. Is there another function that is called as a rare up, before this function is called for each item in the dataset?

Reply:
I don't know of another obvious place to do this, unfortunately.. I'll leave it out for now.